### PR TITLE
Update docs once dataset scripts transferred to the Hub

### LINKED ISSUE #5135: Update docs once dataset scripts transferred to the Hub
## Describe the bug
As discussed in:
- https://github.com/huggingface/hub-docs/pull/423#pullrequestreview-1146083701

we should update our docs once dataset scripts have been transferred to the Hub (and removed from GitHub):
- #4974

Concretely:
- [x] Datasets on GitHub (legacy): https://huggingface.co/docs/datasets/main/en/share#datasets-on-github-legacy
- [x] ADD_NEW_DATASET: https://github.com/huggingface/datasets/blob/main/ADD_NEW_DATASET.md
- ...

This PR complements the work of:
- #5067

This PR is a follow-up of PRs:
- #3777

CC: @julien-c

### PR DESCRIPTION
**Pull request by albertvillanova**
Todo:
- [x] Update docs:
  - [x] Datasets on GitHub (legacy)
  - [x] Load: offline
  - [x] About dataset load:
    - [x] Maintaining integrity
    - [x] Security
- [x] Update docstrings:
  - [x] Inspect:
    - [x] get_dataset_config_info
    - [x] get_dataset_split_names
  - [x] Load:
    - [x] dataset_module_factory
    - [x] load_dataset_builder
    - [x] load_dataset
- [x] Remove `ADD_NEW_DATASET.md`
- [x] Update `.github/ISSUE_TEMPLATE/config.yml`

Fix #5135.

### CODE REVIEW

File: ADD_NEW_DATASET.md [hunk 1]
```diff
@@ -1,357 +0,0 @@
-# How to add one (or several) new datasets to 🤗 Datasets
```
Comment (lhoestq) [inline ADD_NEW_DATASET.md:1]: I think some links are still pointing to this file. Maybe we can just redirect to the Hub or the doc page on how to add a dataset ?

**Review by lhoestq — APPROVED** [review]
Thanks !

File: ADD_NEW_DATASET.md (hunk 1 above)
Comment (albertvillanova) [reply ADD_NEW_DATASET.md:1]: Normally, I changed all the links to this file...

Comment (albertvillanova) [reply ADD_NEW_DATASET.md:1]: I have readded the file with just the links to our docs do add datasets to the Hub: 6c58778

**Pull request merged**